c7n/resources/kms.py

```diff
@@ -62,11 +62,15 @@ def augment(self, resources):
         for r in resources:
             try:
                 key_id = r.get('KeyId')
-                key_arn = r.get('KeyArn', key_id)
-                info = client.describe_key(KeyId=key_arn)['KeyMetadata']
+                # We get `KeyArn` from list_keys and `Arn` from describe_key.
+                # If we already have describe_key info we don't need to fetch
+                # it again.
+                if 'Arn' not in r:
```

Comment:
Given the issue mentioned below, I would argue we should just copy `KeyArn` to `Arn` if it's not set and use a different key (e.g. `CreationDate`) to decide if a describe needs to happen.

Reply:
I'm working on an update to this (following the comments in #6666). I was waffling between copying `KeyArn` to `Arn` or reverting [this change](https://github.com/cloud-custodian/cloud-custodian/commit/05b7d2b95e9f879b2d94b644160630f0c04e923e#diff-c7102cf12e56c1b8e457c58629406c02b8eebfcd6e0ed7072c201fcfd713b008L71-R92).

I'm leaning toward copying `KeyArn` to `Arn` while handling the `AccessDenied` case (since otherwise it comes over following a successful describe).